src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -233,4 +284,29 @@ write(uint64_t offset, const void* buf, uint64_t len) const
   throw error(-ENODEV, "write failed");
 }
 
+
+void 
+device_linux::
+reset(const char* subdev, const char* key, const char* value) const 
+{
+  int val = atoi(value);
+
+  switch(val) {
+  case 1:
```

Comment:
The current code flow here has issues.  It goes from a "string" to an "enum" to a "string" to a "numeric" value.   

I do understand that you were not the original author here.  It would be most helpful if you did refactor the code so that to use enumeration values,  I do realize that this will impact several areas of the code base when you do.

Reply:
hi Stephen, 
i refactored the code.
Please review.